### PR TITLE
Accessibility keyboard navigation to side bar menu and page menu

### PR DESCRIPTION
**Pull request by bolonio**
## Context

While working on a new Accessibility documentation platform I've realized that there are some accessibility issues in the theme. 

This PR just fixes keyboard navigation in the menus.

## Related Issues

- https://github.com/github/accessibility/discussions/2042

### CODE REVIEW

**Review by camertron — APPROVED** [review]

File: theme/src/components/nav-items.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
+import {FocusKeys} from '@primer/behaviors'
 import {LinkExternalIcon} from '@primer/octicons-react'
-import {NavList} from '@primer/react/drafts'
+import {NavList,useFocusZone} from '@primer/react'
```
Comment (joshblack) [inline theme/src/components/nav-items.js:3]: I _think_ we'll need to bump the version of `@primer/react` in `theme` to a newer version as right now it's at `35.2.2` and I think this is only available in later versions

File: theme/src/components/nav-items.js [hunk 2]
```diff
@@ -28,8 +29,9 @@ function NavItem({href, children}) {
 }
 
 function NavItems({items}) {
+  const {containerRef} = useFocusZone({bindKeys: FocusKeys.ArrowVertical | FocusKeys.HomeAndEnd})
```
Comment (joshblack) [inline theme/src/components/nav-items.js:32]: Random question, hope this makes sense 👀 Is there an expectation in a `role="navigation"` context to navigate via a roving tab index? I would have expected this more in an ARIA Menu navigation scheme where-as `role="navigation"` would use tab stops, or maybe the ability to use either scheme.

Separately, when using roving tab, would we need to indicate orientation in some way to communicate that Arrow Up/Down should be used?

File: theme/src/components/nav-items.js (hunk 1 above)
Comment (bolonio) [reply theme/src/components/nav-items.js:3]: That would be ideal, I did it locally and the update breaks some stuff that needs to be adapted. The changes on this PR are independent on which version for `>=35.2.2`. Don't take me wrong, the version should be updated, but not related with the work in this PR.

File: theme/src/components/nav-items.js (hunk 2 above)
Comment (bolonio) [reply theme/src/components/nav-items.js:32]: I've researching about it and I have to say that you're absolutely right. I think I know my mistake. I initially thought about the component as a listbox menu with their own roles, but slowly I was deleting roles, and I totally overseen this use case. Thanks for the review @joshblack :) I will close the PR then.

**Comment by bolonio** [comment]
The `<nav>` element has already `role="navigation"` by default, and no other roles are apply to the side navigation, so navigation over the elements of the list is done using the `Tab Key` and not the `Arrow Up/Down Key`.

**Pull request closed without merging**